src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
```

Comment:
I would make the line breaks more concise this way, if the linter doesn't object:
```suggestion
 ? people.filter(
  p => p.sex === 'm' && people.some(m => m.name === p.mother)
) : people.filter(
  p => people.some(m => m.name === p.mother)
);
```